fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractLoanScheduleGenerator.java

```diff
@@ -430,6 +435,30 @@ private LoanScheduleModel generate(final MathContext mc, final LoanApplicationTe
                 totalOutstanding);
     }
 
+    private void checkDownPaymentInstallmentForOverlappingInstallments(Collection<LoanScheduleModelPeriod> periods,
+            LocalDate scheduledDueDate) {
+        Integer downPaymentPeriodNumber = 0;
+        Integer repaymentPeriodNumber = 0;
+        List<LoanScheduleModelPeriod> overlappingPeriods = periods.stream()
+                .filter(p -> p.periodDueDate() != null && p.periodDueDate().isEqual(scheduledDueDate))
+                .sorted(Comparator.comparing(LoanScheduleModelPeriod::periodNumber)).toList();
+        if (overlappingPeriods.size() == 2) {
```

Comment:
What if there are multiple overlapping periods?

I found this logic a little bit overcomplicated...

You should fetch all overlapping periods (there could be multiple):
- first repayment period, and set the first periodnumber to it
- find and update downpayment period numbers

Reply:
Hi Adam, here i am finding overlapping periods for every period in the loop and checking by comparing with scheduleDueDate. This takes care of multiple overlapping periods.

Reply:
For one single period there will be only two overlapping installments one downpayment and one regular. Since this check is  done for each period in the loop it will take care of overlapping installments for each period .

Reply:
what would happen if i disburse two times or 3 times on the same day as the due date? :)

Reply:
Yes Multiple disbursements on the same due date will not be covered here. I will make changes as per your suggestion.

Reply:
hi Adam, I checked multiple disbursements on same date scenario for overlapping installments and for this case also there are only two overlapping installments are getting created for a given period with the amounts getting updated with  adding new disbursement amount. I added a test for this scenario in LoanAccountPaymentAllocationWithOverlappingDownPaymentInstallmentTest